cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -47,6 +47,9 @@ type InitConfiguration struct {
 
 	// APIEndpoint represents the endpoint of the instance of the API server to be deployed on this node.
 	APIEndpoint APIEndpoint `json:"apiEndpoint,omitempty"`
+
+	// Timeouts defines maximum time intervals kubeadm should wait for completion of critical steps of the init workflow.
+	Timeouts map[KubeadmTimeout]metav1.Duration
```

Comment:
i think this way of mapping the timeouts will work fine and it would be easy to use from YAML.
+1